ee/app/livechat-enterprise/server/business-hour/Multiple.ts

```diff
@@ -0,0 +1,221 @@
+				type: 1,
+			},
+		});
+		for (const businessHour of businessHours) {
+			this.openBusinessHour(businessHour);
+		}
+	}
+
+	async saveBusinessHour(businessHourData: IBusinessHoursExtraProperties): Promise<void> {
+		businessHourData.timezone = {
+			name: businessHourData.timezoneName,
+			utc: this.getUTCFromTimezone(businessHourData.timezoneName),
+		};
+		if (businessHourData.timezone.name) {
+			businessHourData.workHours.forEach((hour: any) => {
+				const startUtc = moment.tz(`${ hour.day }:${ hour.start }`, 'dddd:HH:mm', businessHourData.timezoneName).utc();
+				const finishUtc = moment.tz(`${ hour.day }:${ hour.finish }`, 'dddd:HH:mm', businessHourData.timezoneName).utc();
+				hour.start = {
+					time: hour.start,
+					utc: {
+						dayOfWeek: startUtc.clone().format('dddd'),
+						time: startUtc.clone().format('HH:mm'),
+					},
+					cron: {
+						dayOfWeek: moment(startUtc.format('dddd:HH:mm'), 'dddd:HH:mm').add(moment().utcOffset() / 60, 'hours').format('dddd'),
+						time: moment(startUtc.format('dddd:HH:mm'), 'dddd:HH:mm').add(moment().utcOffset() / 60, 'hours').format('HH:mm'),
+					},
+				};
+				hour.finish = {
+					time: hour.finish,
+					utc: {
+						dayOfWeek: finishUtc.clone().format('dddd'),
+						time: finishUtc.clone().format('HH:mm'),
+					},
+					cron: {
+						dayOfWeek: moment(finishUtc.format('dddd:HH:mm'), 'dddd:HH:mm').add(moment().utcOffset() / 60, 'hours').format('dddd'),
+						time: moment(finishUtc.format('dddd:HH:mm'), 'dddd:HH:mm').add(moment().utcOffset() / 60, 'hours').format('HH:mm'),
+					},
+				};
+			});
+		} else {
+			businessHourData.workHours.forEach((hour: any) => {
+				hour.start = {
+					time: hour.start,
+					utc: {
+						dayOfWeek: moment(`${ hour.day }:${ hour.start }`, 'dddd:HH:mm').utc().format('dddd'),
+						time: moment(`${ hour.day }:${ hour.start }`, 'dddd:HH:mm').utc().format('HH:mm'),
+					},
+					cron: {
+						dayOfWeek: moment(`${ hour.day }:${ hour.start }`, 'dddd:HH:mm').format('dddd'),
+						time: moment(`${ hour.day }:${ hour.start }`, 'dddd:HH:mm').format('HH:mm'),
+					},
+				};
+				hour.finish = {
+					time: hour.finish,
+					utc: {
+						dayOfWeek: moment(`${ hour.day }:${ hour.finish }`, 'dddd:HH:mm').utc().format('dddd'),
+						time: moment(`${ hour.day }:${ hour.finish }`, 'dddd:HH:mm').utc().format('HH:mm'),
+					},
+					cron: {
+						dayOfWeek: moment(`${ hour.day }:${ hour.finish }`, 'dddd:HH:mm').format('dddd'),
+						time: moment(`${ hour.day }:${ hour.finish }`, 'dddd:HH:mm').format('HH:mm'),
+					},
+				};
+			});
+		}
+		businessHourData.active = Boolean(businessHourData.active);
+		businessHourData.type = businessHourData.type || LivechatBussinessHourTypes.MULTIPLE;
+		const departments = businessHourData.departmentsToApplyBusinessHour?.split(',');
+		delete businessHourData.timezoneName;
+		delete businessHourData.departmentsToApplyBusinessHour;
+		delete businessHourData.departments;
+		if (businessHourData._id) {
```

Comment:
It's possible to improve this part of the code, there are some repeated codes that could be simpler.